### DISCUSSION TITLE
ReForge vs A1111

### DISCUSSION THREAD

**Discussion by gohan2091** [Panchovix/stable-diffusion-webui-reForge#92] [Q&A]
I am in the situation where I am still using the original Forge. I want to update. Which should I go to? The latest version of A1111 or should I jump to Reforge? I understand I am on a Reforge discussion page and that opinions may be bias but I just want to know which is the most complete and hear the pros and cons of each.

Thanks

**Comment by Bullseye110** [comment]
I used A1111, Forge, ReForge and Fooocus, so I know kinda well what things work and which don't.
The latest updates from Forge broke it. The older extensions aren't working and the image gen seems laggy.

If you have a weak GPU, use ReForge, but you have to know that the integrated ControlNet is not as good as the ControlNet from A1111.
If you have a good GPU, use A1111 for now as it's best especially if you are using multiple extensions.

**Reply by gohan2091** [reply]
> I used A1111, Forge, ReForge and Fooocus, so I know kinda well what things work and which don't. The latest updates from Forge broke it. The older extensions aren't working and the image gen seems laggy.
> 
> If you have a weak GPU, use ReForge, but you have to know that the integrated ControlNet is not as good as the ControlNet from A1111. If you have a good GPU, use A1111 for now as it's best especially if you are using multiple extensions.

Thanks friend :), I have a 4090 GPU. I will go with A1111 then. I have never used A1111 before (Was on SDNext, then Forge). I hear the backend coding is a mess but I guess if it works that's all that matters.

**Reply by Panchovix** [reply]
Just take in mind that reforge has some features that A1111 doesn't, so it also depends on how much you need certain controlnet new updates (which I plan to upstream, but it is really hard!)

Like new samplers, HiDiffusion, multiple lora views, etc. 

I also have a 4090, and yes I know I did fork Forge to reForge but I preferred Forge over A1111  before as well because I use high size LoRAs, and the difference in performance for that specific case is pretty big, more if moving between models, etc.

**Reply by gohan2091** [reply]
> Just take in mind that reforge has some features that A1111 doesn't, so it also depends on how much you need certain controlnet new updates (which I plan to upstream, but it is really hard!)
> 
> Like new samplers, HiDiffusion, multiple lora views, etc.
> 
> I also have a 4090, and yes I know I did fork Forge to reForge but I preferred Forge over A1111 before as well because I use high size LoRAs, and the difference in performance for that specific case is pretty big, more if moving between models, etc.

When you say new control net do you mean control net union? or the multi-image tab? Also I don't see Reforge on Stability Matrix, will it be added?

**Reply by Panchovix** [reply]
Control net union should semi-work on dev_upstream, but I can't confirm for all cases. Multi image tab I think yeah it's missing.

It seems they're adding support for reForge, you can check https://github.com/LykosAI/StabilityMatrix/issues/760

**Reply by gohan2091** [reply]
> Control net union should semi-work on dev_upstream, but I can't confirm for all cases. Multi image tab I think yeah it's missing.
> 
> It seems they're adding support for reForge, you can check [LykosAI/StabilityMatrix#760](https://github.com/LykosAI/StabilityMatrix/issues/760)

It is the multi image tab I really want as I hear it's very helpful for IPAdapter face V2? Currently I use 3 Control Net IPAdapters at reduced strength. I hear the multi image tab is a better way. This is one of the main reasons why I am considering A1111. I do like the Latent Forge extension though (it's like Latent Couple) so I assume I won't get that on A1111.

**Reply by Panchovix** [reply]
Oh yeah, IMO if you use a lot of controlnet, A1111 is the way for now. I just I knew more about controlnet because I would have updated the extension, but I find it really hard haha

**Reply by gohan2091** [reply]
> Oh yeah, IMO if you use a lot of controlnet, A1111 is the way for now. I just I knew more about controlnet because I would have updated the extension, but I find it really hard haha

I see lol, to be honest, I don't understand why ReForge and A1111 exists separately. Why can't they exist as one package?

**Reply by Panchovix** [reply]
A1111 has his own backend and merging the reforge (aka comfy) backend into it is really hard and time consuming.

**Reply by Bullseye110** [reply]
> > Control net union should semi-work on dev_upstream, but I can't confirm for all cases. Multi image tab I think yeah it's missing.
> > It seems they're adding support for reForge, you can check [LykosAI/StabilityMatrix#760](https://github.com/LykosAI/StabilityMatrix/issues/760)
> 
> It is the multi image tab I really want as I hear it's very helpful for IPAdapter face V2? Currently I use 3 Control Net IPAdapters at reduced strength. I hear the multi image tab is a better way. This is one of the main reasons why I am considering A1111. I do like the Latent Forge extension though (it's like Latent Couple) so I assume I won't get that on A1111.

One more thing. I did some more experiments yesterday and here's what I found:
1. - Somehow (I'm no coder, so I don't really know why or how) the images generated with the latest commit of A1111 are far superior in quality to ReForge -> tried only with pony diffusion model.
2. - Using 2-3 or more ControlNets on ReForge will give you an error. Using 1 ControlNet with Hires. Fix or with Kohya Hires. Fix on ReForge also gives you errors. A1111 doesn't have any of these problems, but it requires more VRAM for each ControlNet (ControlNet Union and Union Promax work well on A1111 and decent on ReForge).

**Comment by Bullseye110** [comment]
Gonna retest some features on the latest ReForge version to see if it's better than A1111 for now, but I guess that it won't really be until ControlNet will be updated.

**Reply by Panchovix** [reply]
Updated yesterday to some controlnet upstream preprocessor, but the extension itself is probably still missing features.

**Reply by Bullseye110** [reply]
Maybe you can figure something out in the forge branch on which you cloned. A problem with the newest official Forge commit is that it makes this extension (https://github.com/viyiviyi/sd-encrypt-image - the working commit of this extension was 376358d giving 100% artifact images with strange lines, dots and colors) to not work anymore... And this extension is very useful especially for the cloud GPU users.
